src/core/etl/src/Flow/ETL/Row/Reference/Expression/Cast/XMLConverter.php

```diff
@@ -0,0 +1,92 @@
+<?php declare(strict_types=1);
+
+namespace Flow\ETL\Row\Reference\Expression\Cast;
+
+final class XMLConverter
+{
+    /**
+     * @param \DOMDocument $document
+     *
+     * @return array<mixed>
+     */
+    public function toArray(\DOMDocument $document) : array
+    {
+        $xmlArray = [];
+
+        if ($document->hasChildNodes()) {
+            $children = $document->childNodes;
+
+            foreach ($children as $child) {
```

Comment:
IMO `$children` variable is useless here and `$child` should be renamed to `$node`